contracts/augur/trading/ZeroXTrade.sol

```diff
@@ -191,26 +48,31 @@ contract ZeroXTrade is Initializable, IZeroXTrade, IERC1155 {
         address _affiliateAddress,
         bytes32 _tradeGroupId,
         IExchange.Order[] memory _orders,
-        bytes[] memory _signatures
+        bytes[] memory _signatures,
+        address _taker
     )
         public
         payable
         returns (uint256)
     {
         uint256 _fillAmountRemaining = _requestedFillAmount;
 
-        transferFromAllowed = true;
+        // transferFromAllowed = true;
 
-        uint256 _protocolFee = 150000 * tx.gasprice;
+        // uint256 _protocolFee = 150000 * tx.gasprice;
 
         // Do the actual asset exchanges
         for (uint256 i = 0; i < _orders.length && _fillAmountRemaining != 0; i++) {
             IExchange.Order memory _order = _orders[i];
-            validateOrder(_order);
-            IExchange _exchange = getExchangeFromAssetData(_order.makerAssetData);
+            // @discuss not sure what validations are required for orders that were signed for Matic and how to retrieve _exchange
+            // validateOrder(_order);
+
+            // @discuss How to get _exchange address from makerAssetData for an order that was created on Matic
```

Comment:
**_Discuss:_** How to get _exchange address from makerAssetData for an order that was created on Matic

One solution is, I can have a mapping on the root chain which maps `maticExchangeAddress => maticExchange contract address on rootchain`.
```
IExchange _exchange = getExchangeFromAssetData(_order.makerAssetData);

function getExchangeFromAssetData(makerAssetData) returns (_rootExchangeContract) {
    exchange := mload(add(_assetData, 160))
    _rootExchangeContract = maticToRootExchangeMapping(exchange);
    require(_rootExchangeContract != address(0));
}
```

Reply:
Why can't we just get it form the assetData as we did before?